recursion.rb

```diff
@@ -1,14 +1,48 @@
-
+#Recurisive cas: n! = n *(n-1)!
+#Basecase: 1! = 1
+#Basecase: 0! = 1
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+  return n * fact(n-1)
 end
 
+# fib(1) = 1
+# fib(2) = 1
+# fib(3) = 2
+# fib(3) = fib(2) + fib(1)
+# fib(4) = fib(3) + fib(2)
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n-1) + fib(n-2)
 end
 
 def pal(s)
+  if s.length == 1 || s.length == 0
+    return true
+  end
+  if s[0] == s[-1]
+     pal(s[1..-2])
+  else
+    return false
+  end
 end
```

Comment:
Good solution. I like how you handled the recursive case; it is simple and uncomplicated (well, as simple and uncomplicated as a working solution can be). Readable and nicely done.